p2p/node/node.go

```diff
@@ -91,6 +92,12 @@ func NewNode(ctx context.Context) (*P2PNode, error) {
 		return nil, err
 	}
 
+	// create a DHT storage instance
+	ds, err := dhtstorage.NewDHTStorage()
+	if err != nil {
+		log.Global.Fatalf("error creating DHT storage: %s", err)
```

Comment:
If err != nil then datastore will be nil. If we still store the datastore in libp2p then I expect this to cause a panic so we should return err here.

Reply:
Oh I overlooked that it was a `Fatal` error. That makes sense